add more defaults to test/kwok/cluster pipeline

kranurag7 opened the pull request:
we duplicate these parameters over and over again in package tests. Adding these here given these are good defaults which always remains the same.

some packages where these are getting repeated are:
```bash
$ rg 'KUBERNETES_SERVICE_PORT' -l
mongodb-kubernetes-operator.yaml
kube-state-metrics.yaml
cluster-proportional-autoscaler.yaml
sealed-secrets.yaml
kyverno-1.13.yaml
flux-source-controller.yaml
lvm-driver.yaml
ruby3.3-fluentd-kubernetes-daemonset-1.18.yaml
kubernetes-event-exporter.yaml
ruby3.2-fluentd-kubernetes-daemonset-1.18.yaml
ruby3.1-fluentd-kubernetes-daemonset-1.18.yaml
redis-operator.yaml
ruby3.4-fluentd-kubernetes-daemonset-1.18.yaml
metrics-server.yaml
```

kranurag7 committed:
add more defaults to test/kwok/cluster pipeline

we duplicate these parameters over and over again in package tests.
Adding these here given these are good defaults which always remains the
same.

Signed-off-by: kranurag7 <81210977+kranurag7@users.noreply.github.com>

maxgio92 commented on pipelines/test/kwok/cluster.yaml:
```diff
@@ -45,3 +51,11 @@ pipeline:
+      CA=$(kubectl config view --raw -o jsonpath='{.clusters[0].cluster.certificate-authority}')
+      cp $CA /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+      echo ${{inputs.namespace}} > /var/run/secrets/kubernetes.io/serviceaccount/namespace
+      kubectl create sa ${{inputs.namespace}}
```
```suggestion
      kubectl create sa default
```

maxgio92 commented on pipelines/test/kwok/cluster.yaml:
```diff
@@ -45,3 +51,11 @@ pipeline:
+      cp $CA /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+      echo ${{inputs.namespace}} > /var/run/secrets/kubernetes.io/serviceaccount/namespace
+      kubectl create sa ${{inputs.namespace}}
+      kubectl create token ${{inputs.namespace}} > /var/run/secrets/kubernetes.io/serviceaccount/token
```
I'd not create the token by default following k8s. For security reason, token are not created by default since 1.23, and they're not always required in our tests IMHO

maxgio92 commented on pipelines/test/kwok/cluster.yaml:
```diff
@@ -45,3 +51,11 @@ pipeline:
       kubectl wait --for=condition=Ready nodes --all
       kubectl cluster-info
+      export KUBERNETES_SERVICE_HOST="127.0.0.1"
+      export KUBERNETES_SERVICE_PORT="32764"
```
These will be set only for the current pipeline step, which is only this shell process and children that run kwok.
Something like this will fail:
```
...
  - uses: test/kwok/cluster
  - runs: |
      set -u
      echo $KUBERNETES_SERVICE_HOST
```

with
```
...
2025/04/15 10:56:53 WARN /bin/sh: KUBERNETES_SERVICE_HOST: parameter not set
```

kranurag7 committed:
address max review comments

exports are not required and creating token is security sensitive so
omitting that too.

Signed-off-by: kranurag7 <81210977+kranurag7@users.noreply.github.com>

a-crate commented:
Do all the packages which use this pipeline still pass testing after these changes?

kranurag7 commented on pipelines/test/kwok/cluster.yaml:
```diff
@@ -45,3 +51,7 @@ pipeline:
       kubectl wait --for=condition=Ready nodes --all
       kubectl cluster-info
+      mkdir -p /var/run/secrets/kubernetes.io/serviceaccount
+      CA=$(kubectl config view --raw -o jsonpath='{.clusters[0].cluster.certificate-authority}')
+      cp $CA /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
```
thanks Jamon for reviews, I think if we hardcode this to `/root` then that should work but we might run into issues when we use `accounts:` block inside tests. 

but it'll fail anyways if we use any other UID here because that uid won't have enough permissions to create dirs in `/var/` directory.

So, Yes I think we can do this change.
